web/src/engine/websites/CiaoPlus.ts

```diff
@@ -0,0 +1,213 @@
+import { Tags } from '../Tags';
+import icon from './CiaoPlus.webp';
+import type { MangaPlugin } from '../providers/MangaPlugin';
+import { Chapter, DecoratableMangaScraper, Manga, Page } from '../providers/MangaPlugin';
+import * as Common from './decorators/Common';
+import { FetchJSON } from '../platform/FetchProvider';
+import type { Priority } from '../taskpool/DeferredTask';
+import DeScramble from '../transformers/ImageDescrambler';
+
+type APIMangas = {
+    title_list: {
+        title_id: number,
+        title_name: string
+        episode_id_list: number[]
+    }[]
+
+}
+
+type APIChapters = {
+    episode_list: {
+        episode_id: number,
+        episode_name: string
+    }[]
+}
+
+type APIPages = {
+    scramble_seed: number,
+    page_list: string[]
+}
+
+type TDimension = {
+    width: number,
+    height: number
+}
+
+export default class extends DecoratableMangaScraper {
+
+    private readonly mangaRegexp = new RegExp(`^${this.URI.origin}/comics/title/(\\d+)/episode/\\d+$`);
+    private readonly apiUrl = 'https://api.ciao.shogakukan.co.jp/';
+
+    public constructor() {
+        super('ciaoplus', 'Ciao Plus', 'https://ciao.shogakukan.co.jp', Tags.Media.Manga, Tags.Language.Japanese, Tags.Source.Official);
+    }
+
+    public override get Icon() {
+        return icon;
+    }
+
+    public override ValidateMangaURL(url: string): boolean {
+        return this.mangaRegexp.test(url);
+    }
+
+    public override async FetchManga(provider: MangaPlugin, url: string): Promise<Manga> {
+        const mangaid = parseInt(url.match(this.mangaRegexp)[1]);
+        const endpointUrl = new URL(`/title/list?version=6.0.0&platform=3&title_id_list=${mangaid}`, this.apiUrl);
+        const request = await this.CreateRequest(endpointUrl);
+        const { title_list } = await FetchJSON<APIMangas>(request);
```

Comment:
May spread directly to first array element `const { title_list: [ manga ] }`

Reply:
I was wondering how to spread it, thanks